### ISSUE TITLE
There is the fail "UNIQUE constraint failed" with a LAZY collection

### ISSUE THREAD

**Issue by Sysaninster** [doctrine/doctrine2#6613]
I have the User entity with the LAZY collection field. If I do this code:

```php
        $user = new User();
        $user->id = 1;
        $this->_em->persist($user);
        $this->_em->flush();

        $this->_em->clear();

        /** @var User $user */
        $user = $this->_em->find(User::class, 1);
        $phone1 = new Phone();
        $phone1->id = 1;
        $user->phones->add($phone1);
        $this->_em->persist($phone1);
        $this->_em->flush();

        $phone2 = new Phone();
        $phone2->id = 2;
        $user->phones->add($phone2);
        $this->_em->persist($phone2);
        $user->phones->toArray();
        $this->_em->flush();
```

I will get the error:

```
Doctrine\DBAL\Exception\UniqueConstraintViolationException: An exception occurred while executing 'INSERT INTO user_phone (user_id, phone_id) VALUES (?, ?)' with params [1, 1]:

SQLSTATE[23000]: Integrity constraint violation: 19 UNIQUE constraint failed: user_phone.user_id, user_phone.phone_id
```

If I remove the *$user->phones->toArray();* line then it will work without a problem.

**Comment by Ocramius** [comment]
@Sysaninster can you put this into a test case like the ones in https://github.com/doctrine/doctrine2/tree/bbe005837e51442e0d6037bafe70dc4b7712dac1/tests/Doctrine/Tests/ORM/Functional/Ticket please?

**Comment by Ocramius** [comment]
Closing here, thanks for sending a failing test case!

**Comment by Ocramius** [comment]
Ref: #6614

**Comment by Ocramius** [comment]
Reopening, since this will be closed once #6616 is merged.

**Comment by Ocramius** [comment]
@Sysaninster thanks for the bug report - this is fixed and will go into `2.5.7` ASAP
